app/models/invoice.rb

```diff
@@ -3,6 +3,19 @@ class Invoice < ApplicationRecord
   belongs_to :merchant
 
   has_many :transactions
+  has_many :invoice_items
 
   validates :status, presence: true
+
+  def total
+    invoice_items.reduce(0) do |total, item|
+      total += (item.unit_price * item.quantity)
+    end
+  end
+
+  def successful?
```

Comment:
Moving successful results feels tricky to me. If I do it as part my query in merchants, I'll be filtering merchants. I want to make sure that I'm leaving that logic with the object it belongs with, right?